proposals/XXXX-extend-property-wrappers-to-functions-and-closures.md

```diff
@@ -0,0 +1,391 @@
+# Extend Property Wrappers to Functions and Closures
```

Comment:
Nit: Let's change the title slightly to "Extend Property Wrappers to Function and Closure Parameters" to clarify that we aren't proposing property wrappers on functions themselves